### ISSUE TITLE
The loss function in mat based methods

### ISSUE THREAD

**Issue by ATLAS8346** [hzzheng93/ATTA#1]
The loss function used in the ``mat'' based methods is weird. Why the input of the CrossEntropyLoss() is the logit after log_sofmax() rather than the logit itself?

**Comment by neuer93** [comment]
Hi, the `log_softmax()` could be redundant there. In pytorch implementation, celoss will automatically apply log_softmax if the input data is not log_softmax. 

Try this example:
```
import torch.nn.functional as F
import torch

loss = nn.CrossEntropyLoss()

predicted = torch.randn(16,10)
targeted = torch.ones(16).long()
predicted_soft = F.log_softmax(predicted)

print(targeted)
print(predicted[:3])
print(predicted_soft[:3])

print(loss(predicted, targeted))
print(loss(predicted_soft, targeted))
```

Output:
```
tensor([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])
tensor([[ 0.5095,  0.1436, -0.8504,  2.4353, -0.0722, -0.2957,  1.3351,  1.4147,
         -0.6626, -0.1438],
        [ 0.2922,  1.0302,  0.1522,  0.2687,  1.0356,  0.4665,  1.0093,  0.4876,
         -0.0136,  1.2093],
        [-0.9302, -0.1674, -0.5489,  0.7785, -0.2676, -1.5789, -1.0356, -0.7717,
         -2.0956, -0.4747]])
tensor([[-2.7345, -3.1004, -4.0944, -0.8087, -3.3163, -3.5397, -1.9089, -1.8293,
         -3.9066, -3.3878],
        [-2.6902, -1.9522, -2.8302, -2.7136, -1.9467, -2.5159, -1.9730, -2.4948,
         -2.9960, -1.7731],
        [-2.8065, -2.0438, -2.4253, -1.0979, -2.1440, -3.4552, -2.9120, -2.6481,
         -3.9720, -2.3510]])
tensor(2.7762)
tensor(2.7762)
```

**Comment by ATLAS8346** [comment]
Hi, thanks for your quick response. I have corrected the loss function of your implementation. Although the final performance slightly differs from the original code, the proposed method still enjoys great improvement in terms of the efficiency. I thus will close this issue.
